packages/api/src/derives/balances.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Adding this file to support extrinsic via rv36
+import { ApiInterfaceRx } from '@polkadot/api/types';
+import { Observable, of, combineLatest } from 'rxjs';
+import { DeriveBalancesAccount } from '@cennznet/api/derives/types';
+import { map, switchMap } from 'rxjs/operators';
+import { memo } from '@polkadot/api-derive/util';
+import { AccountId, Index } from '@cennznet/types';
+import { AccountData, AccountInfo } from '@polkadot/types/interfaces';
+import { ITuple } from '@polkadot/types/types';
+import { isFunction } from '@polkadot/util';
+
+function calcBalances(api: ApiInterfaceRx, [accountId, accountNonce]: [AccountId, Index]): DeriveBalancesAccount {
+  return {
+    accountId,
+    accountNonce,
+  };
+}
+// With runtime version <= 36, the way nonce is queried is as follows
+function queryOldNonce(api: ApiInterfaceRx, accountId: AccountId): Observable<Index> {
+  return api.query.system.accountNonce<Index>(accountId).pipe(map((accountNonce): Index => accountNonce));
+}
+
+function queryCurrentNonce(api: ApiInterfaceRx, accountId: AccountId): Observable<Index> {
```

Comment:
should we exchange this for `api.rpc.system.accountNextIndex`? not sure that we use AccountData...

Reply:
this works.. have also checked on UI...
<img width="1574" alt="Screen Shot 2020-12-02 at 3 46 36 PM" src="https://user-images.githubusercontent.com/29415595/100821603-92b51d00-34b5-11eb-8ebc-1cacc925ed7a.png">

I am not sure how rpc and rx behaves.. I can give this a try too..

Reply:
I couldn't think of a specific test we have to check this doesn't break anything e.g checking nonce is incrementing correctly by calling this

Reply:
updated..